asmupload.sh

```diff
@@ -30,16 +30,23 @@ if [[ $ERROR_CODE -ne 0 ]]; then
 	exit $ERROR_CODE
 fi
 
+# If .asm is used, strip file extension
+if [ ${2: -4} == ".asm" ]; then
```

Comment:
This isn't fully compatible with macOS, and we should keep that compatible.

macOS'es version of `[` does not support the `==` condition. Bash's built-in operator `[[` does (even in the old version macOS ships). So I would propose to use that.

While you're at it, add some extra quotes around `${2: -4}` as a means of defensive programming, although it should not really matter as `${2: -4}` should never return an empty string.

That should give you: `if [[ "${2: -4}" == ".asm" ]]; then`